packages/meditrak-server/src/routes/dashboardVisualisations/GETDashboardVisualisations.js

```diff
@@ -44,12 +75,38 @@ export class GETDashboardVisualisations extends GETHandler {
     if (!report) {
       throw new Error(`Cannot find report "${dashboardItem.report_code}" of the visualisation`);
     }
-  
+
     const permissionGroup = await this.models.permissionGroup.findById(report.permission_group_id);
 
     return {
-      dashboardItem: mapKeys(dashboardItem, (_, propertyKey) => camel(propertyKey)),
-      report: { code: report.code, config: report.config, permissionGroup: permissionGroup.name },
+      dashboardItem: camelKeys(dashboardItem),
+      report: {
+        code: report.code,
+        config: report.config,
+        permissionGroup: permissionGroup.name,
+      },
     };
   }
+
+  async findRecords(inputCriteria) {
+    const criteria = parseCriteria(inputCriteria);
+
+    const dashboardItems = await this.models.dashboardItem.find(criteria);
+    const reports = await this.models.report.find({ code: dashboardItems.map(di => di.code) });
+    const reportsByCode = keyBy(reports, 'code');
+    const permissionGroups = await this.models.permissionGroup.find({
+      id: reports.map(r => r.permission_group_id),
+    });
+    const permissionGroupsById = keyBy(permissionGroups, 'id');
+
+    return dashboardItems.map(dashboardItem =>
+      buildVisualisationObject(dashboardItem, reportsByCode, permissionGroupsById),
+    );
+  }
+
+  async countRecords(inputCriteria) {
+    return this.database.count(TYPES.DASHBOARD_ITEM, parseCriteria(inputCriteria), {
+      report: ['dashboard_item.code', 'report.code'],
+    });
```

Comment:
It looks like you're wanting to join to `report` here, but I think the syntax is something like:
```suggestion
    return this.database.count(TYPES.DASHBOARD_ITEM, parseCriteria(inputCriteria), {
      joinWith: 'report',
      joinCondition: ['dashboard_item.report_code', 'report.code'],
    });
```
However, maybe it's just not needed at all for a count?

Reply:
Oops good catch!

Reply:
About whether it's needed: I used the implicit definition of a "dashboard visualisation" being a combination of a dashboard_item + a report, where both resources must be present.

In my implementation, all requested items should have a report at this point: if not, an error would have been thrown in [L31](https://github.com/beyondessential/tupaia/blob/mel-28-viz-builder-import-endpoint/packages/meditrak-server/src/routes/dashboardVisualisations/GETDashboardVisualisations.js#L31). However, in the interest of clarity and robustness in the face of potential future refactoring, I'm thinking of keeping the join here - let me know if you disagree.

Reply:
Happy to keep it :-)